proto/cosmos/gov/query.proto

```diff
@@ -0,0 +1,151 @@
+  rpc Deposit(QueryDepositRequest) returns (QueryDepositResponse) {}
+
+  // Deposits returns all deposits of a single proposal
+  rpc Deposits(QueryDepositsRequest) returns (QueryDepositsResponse) {}
+
+  // TallyResult queries the tally of a proposal vote
+  rpc TallyResult(QueryTallyRequest) returns (QueryTallyResponse) {}
+}
+
+// QueryProposalRequest is the request type for the Query/Proposal RPC method
+message QueryProposalRequest {
+  // unique id of the proposal
+  uint64 proposal_id = 1;
+
+  cosmos.query.PageRequest req = 2;
+}
+
+// QueryProposalResponse is the response type for the Query/Proposal RPC method
+message QueryProposalResponse {
+  cosmos.gov.Proposal proposal = 1 [(gogoproto.nullable) = false];
+}
+
+// QueryProposalsRequest is the request type for the Query/Proposals RPC method
+message QueryProposalsRequest {
+  // status of the proposals.
+  int32 proposal_status = 1;
+  
+  // Voter address for the proposals.
+  bytes voter = 2 [(gogoproto.casttype) = "github.com/cosmos/cosmos-sdk/types.AccAddress"]; 
+
+  // Deposit addresses from the proposals.
+  bytes depositor = 3 [(gogoproto.casttype) = "github.com/cosmos/cosmos-sdk/types.AccAddress"];
+
+  cosmos.query.PageRequest req = 4;
+}
+
+// QueryProposalsResponse is the response type for the Query/Proposals RPC method
+message QueryProposalsResponse {
+  repeated cosmos.gov.Proposal proposals = 1 [(gogoproto.nullable) = false];
+
+  cosmos.query.PageResponse res = 2;
+}
+
+// QueryVoteRequest is the request type for the Query/Vote RPC method
+message QueryVoteRequest {
+  // unique id of the proposal
+  uint64 proposal_id = 1;
+  
+  // Voter address for the proposals.
+  bytes voter = 2 [(gogoproto.casttype) = "github.com/cosmos/cosmos-sdk/types.AccAddress"]; 
+}
+
+// QueryVoteResponse is the response type for the Query/Vote RPC method
+message QueryVoteResponse {
+  cosmos.gov.Vote vote = 1 [(gogoproto.nullable) = false];
+}
+
+// QueryVotesRequest is the request type for the Query/Votes RPC method
+message QueryVotesRequest {
+  // unique id of the proposal
+  uint64 proposal_id = 1;
+
+  cosmos.query.PageRequest req = 2;
+}
+
+// QueryVotesResponse is the response type for the Query/Votes RPC method
+message QueryVotesResponse {
+  repeated cosmos.gov.Vote votes = 1 [(gogoproto.nullable) = false];
+
+  cosmos.query.PageResponse res = 2;
+}
+
+// QueryParamsRequest is the request type for the Query/Params RPC method
+message QueryParamsRequest { 
+  string params_type = 1;
+}
+
+// QueryParamsResponse is the response type for the Query/Params RPC method
+message QueryParamsResponse {
+  cosmos.gov.VotingParams VotingParams = 1 [(gogoproto.customtype) = "VotingParams", (gogoproto.nullable) = false];
+  cosmos.gov.DepositParams DepositParams = 2 [(gogoproto.customtype) = "DepositParams", (gogoproto.nullable) = false];
+  cosmos.gov.TallyParams TallyParams = 3 [(gogoproto.customtype) = "TallyParams", (gogoproto.nullable) = false];
```

Comment:
Does this need to be prefixed with `cosmos.gov`?

Reply:
Not exactly needed, updated